tests/h/services/search_index/_queue_test.py

```diff
@@ -137,206 +136,192 @@ def add_where(self, queue):
-        self, annotations, annotation_ids, batch_indexer, now, queue
+        self, batch_indexer, factories, now, queue, SyncAnnotationJobFactory
     ):
-        # Simulate the annotations having been updated in the DB after they
-        # were indexed.
-        for annotation in annotations:
-            annotation.updated = now
+        annotation = factories.Annotation()
+        SyncAnnotationJobFactory(annotation=annotation, index=True)
+        # Simulate the annotation having been updated in the DB after it was
+        # indexed.
+        annotation.updated = now
 
-        counts = queue.sync(LIMIT)
+        counts = queue.sync(1)
 
         assert counts == {
-            Queue.Result.SYNCED_DIFFERENT: LIMIT,
-            Queue.Result.SYNCED_TOTAL: LIMIT,
+            Queue.Result.SYNCED_DIFFERENT: 1,
+            Queue.Result.SYNCED_TOTAL: 1,
         }
-        batch_indexer.index.assert_called_once_with(
-            Any.list.containing(annotation_ids).only()
-        )
+        batch_indexer.index.assert_called_once_with([annotation.id])
 
-    @pytest.mark.usefixtures("with_indexed_annotations", "with_queued_annotations")
     def test_if_the_annotation_has_a_different_userid_in_Elastic_it_indexes_it(
-        self, annotations, annotation_ids, batch_indexer, queue
+        self, batch_indexer, factories, queue, SyncAnnotationJobFactory
     ):
+        annotation = factories.Annotation()
+        SyncAnnotationJobFactory(annotation=annotation, index=True)
         # Simulate the user having been renamed in the DB.
-        for annotation in annotations:
-            annotation.userid = "new_userid"
+        annotation.userid = "new_userid"
 
-        counts = queue.sync(LIMIT)
+        counts = queue.sync(1)
 
         assert counts == {
-            Queue.Result.SYNCED_DIFFERENT: LIMIT,
-            Queue.Result.SYNCED_TOTAL: LIMIT,
+            Queue.Result.SYNCED_DIFFERENT: 1,
+            Queue.Result.SYNCED_TOTAL: 1,
         }
-        batch_indexer.index.assert_called_once_with(
-            Any.list.containing(annotation_ids).only()
-        )
+        batch_indexer.index.assert_called_once_with([annotation.id])
 
     def test_if_there_are_multiple_jobs_with_the_same_annotation_id(
-        self, annotation_ids, batch_indexer, add_all, queue
+        self, batch_indexer, factories, queue, SyncAnnotationJobFactory
     ):
-        add_all(ids=[annotation_ids[0] for _ in range(LIMIT)])
+        annotation = factories.Annotation()
+        jobs = SyncAnnotationJobFactory.create_batch(size=2, annotation=annotation)
 
-        counts = queue.sync(LIMIT)
+        counts = queue.sync(len(jobs))
 
-        # It only syncs the annotation to Elasticsearch once.
+        # Unfortunately the method gets the metrics wrong here: it reports that
+        # it synced two annotations because it processed two different jobs.
+        # But actually the two jobs were for the same annotation and it only
+        # synced one.
```

Comment:
So our metrics are actually wrong. The overuse of `LIMIT` in the tests was hiding this fact from the eyes, I noticed it when removing `LIMIT`. I'd like to fix this in a follow-up PR.